src/main/java/seedu/address/logic/commands/NewtagCommand.java

```diff
@@ -13,29 +15,30 @@ public class NewtagCommand extends Command {
     public static final String COMMAND_WORD = "newtag";
     public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds a new tag (case insensitive).\n"
             + "Parameters: TAG_NAME (MAX 50 alphanumeric characters, spaces, parenthesis and apostrophes)\n"
-            + "Example: " + COMMAND_WORD + " Bride's Friend";
-    public static final String MESSAGE_SUCCESS = "New tag added: ";
-    public static final String MESSAGE_DUPLICATE = "This tag already exists.\n";
-    private final Tag tag;
+            + "Example: " + COMMAND_WORD + " t/Bride's Friend";
+    public static final String MESSAGE_SUCCESS = "New tag(s) added: ";
+    public static final String MESSAGE_DUPLICATE = "Some tag(s) provided have been added before.\n";
+    private final List<Tag> tags;
 
     /**
-     * @param tag The tag object to be added.
+     * @param tags The tags to be added.
      */
-    public NewtagCommand(Tag tag) {
-        requireAllNonNull(tag);
-        this.tag = tag;
+    public NewtagCommand(List<Tag> tags) {
+        requireAllNonNull(tags);
+        this.tags = tags;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireAllNonNull(model);
-        boolean isSuccessful = model.addTag(tag);
+        boolean isSuccessful = model.addTags(tags);
         if (!isSuccessful) {
             throw new CommandException(MESSAGE_DUPLICATE);
         }
+        String successMessage = MESSAGE_SUCCESS + " " + tags + "\n";
```

Comment:
I think the extra spacing is unnecessary since your MESSAGE_SUCCESS already ends with an empty space